public/assets/scripts/reusable-elements.js

```diff
@@ -99,6 +108,58 @@ class SearchBar extends HTMLElement {
         const btn = document.querySelector(".search-bar > button");
         btn.onClick = searchFunction;
     }
+
+    filterItems(e) {
+        const query = e.target.value.toLowerCase();
+        const context = this.getAttribute("data-context");
+
+        if (context === "documentation") {
+            this.filterDocs(query);
+        } else if (context === "gallery") {
+            this.filterGallery(query);
+        }
+    }
+
+    filterDocs(query) {
+        // Get all sections with .doc-table-content
+        const tables = document.querySelectorAll(".doc-table-content");
+
+        tables.forEach(table => {
+            let anyVisible = false;
+            const functionRows = table.querySelectorAll('tr');
+
+            functionRows.forEach(row => {
+                const functionNameElement = row.querySelector(".table-keyword");
+                if (functionNameElement) {
+                    const functionName = functionNameElement.textContent.toLowerCase();
+                    const isVisible = functionName.includes(query);
+                    row.style.display = isVisible ? "" : "none";
+                    anyVisible = anyVisible || isVisible;
+                }
+            });
+
+            // Determine the associated .no-match-message for this table.
+            const section = table.closest('section');
+            const messageElement = section.querySelector('.no-match-message');
+            messageElement.style.display = anyVisible ? 'none' : 'block';
```

Comment:
Perhaps putting this part into a helper function is better because similar code also exists in `filterGallery`. However since there's only two lines it's really up to you.